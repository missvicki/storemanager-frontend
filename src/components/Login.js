import React from 'react';
import { Message, Header, Form, Grid, Segment } from 'semantic-ui-react';

import '../styles/login.css'

const Login = (props) => {
    const {
        user_name,
        password,
        loginError,
        onSubmitHandler,
        onChangeHandler,
        onDismissHandler,
        loading
    } = props
    const loginErorMsg = (
        <Message
            negative
            content={loginError}
            onDismiss={onDismissHandler}
            style={{ width: "50%", margin: "auto", color: "red" }}
        />
    );

    return (
        <div style={{ padding: "10% 0" }} className="loginPage">
            <Grid columns={2} stackable>
                <Grid.Column width={4}></Grid.Column>
                <Grid.Column width={8} padded="very">
                    <Segment padded="very" color="teal">
                        <Header as="h1" textAlign="center" padded="very">Store Manager</Header>
                        <Header as="h6" style={{ color: "#008080", marginTop: "5px", textAlign: "center", fontSize: "1em" }}>
                            Log in to your account
                        </Header>
                        <br />
                        {loginError ? loginErorMsg : null}
                        <br />
                        <center>
                            <Form onSubmit={onSubmitHandler}>
                                <Form.Field>
                                    <input
                                        type="text"
                                        placeholder="Username"
                                        name="user_name"
                                        value={user_name}
                                        onChange={onChangeHandler}
                                    />
                                </Form.Field>
                                <Form.Field>
                                    <input
                                        type="password"
                                        placeholder="Password"
                                        name="password"
                                        value={password}
                                        onChange={onChangeHandler}
                                    />
                                </Form.Field>
                                <Form.Button type="submit" style={{ background: "teal", color: "white" }} loading={loading}>Sign in</Form.Button>
                            </Form>
                        </center>
                    </Segment>

                </Grid.Column>
            </Grid>

        </div>
    )
};

export default Login;
