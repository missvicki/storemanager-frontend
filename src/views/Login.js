import React, { Component } from 'react';
import { connect } from "react-redux";

import Login from '../components/Login';
import loginFetch from "../actions/loginAction"
import { removeLoginError } from "../actions/index";

export class LoginView extends Component {
    constructor(props) {
        super(props);
        this.state = {
            user_name: "",
            password: "",
            role: "admin"
        };
    }

    onSubmitHandler = e => {
        e.preventDefault();
        const { user_name, password, role } = this.state;
        const data = { user_name, password, role };
        const url =
            "https://store-manager-ap1.herokuapp.com/api/v2/auth/login";
        const payload = {
            data,
            history: this.props.history,
            url
        };
        this.props.login(payload);
    };

    onDismissHandler = () => {
        const {
            dismissLoginError
        } = this.props
        dismissLoginError();
    };

    onChangeHandler = e => this.setState({ [e.target.name]: e.target.value });

    render() {
        const loginProps = {
            user_name: this.state.user_name,
            password: this.state.password,
            role: this.state.role,
            loginError: this.props.loginError,
            onSubmitHandler: this.onSubmitHandler,
            onChangeHandler: this.onChangeHandler,
            onDismissHandler: this.onDismissHandler,
            loading: this.props.loading
        };
        return (
            <Login {...loginProps} />
        );
    };
};

export const mapDispatchToProps = dispatch => {
    return {
        login: payload => dispatch(loginFetch(payload)),
        dismissLoginError: () => dispatch(removeLoginError())
    };
};

export const mapStateToProps = state => {
    return {
        loginError: state.login.loginError,
        loading: state.login.loading
    };
};

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(LoginView);
