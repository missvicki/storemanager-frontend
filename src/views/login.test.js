import React from "react";
import { shallow } from "enzyme";
import { LoginView, mapDispatchToProps, mapStateToProps } from "./Login";

describe('login view', () => {
    const mockLoginFetch = jest.fn();
    const mockRemoveLoginError = jest.fn();
    const dispatch = jest.fn();

    const loginProps = {
        login: mockLoginFetch,
        dismissLoginError: mockRemoveLoginError,
        loading: false,
        loginError: ""
    };

    const loginView = shallow(<LoginView {...loginProps} />);
    const loginViewInstance = loginView.instance();

    afterEach(() => {
        loginView.setState({ user_name: "", password: "", role: "admin" });
    });

    it("renders loginView component correctly", () => {
        expect(loginView).toMatchSnapshot();
    });

    it("onChangeHandler updates user_name in state when an event with value of user_name is passed", () => {
        loginViewInstance.onChangeHandler({
            target: { value: "victoria", name: "user_name" }
        });
        expect(loginView.state().user_name).toBe("victoria");
    });

    it("onChangeHandler updates password in state when an event with value of password is passed", () => {
        loginViewInstance.onChangeHandler({
            target: { value: "okay", name: "password" }
        });
        expect(loginView.state().password).toBe("okay");
    });

    it("onDismissHandler dispatches removeLoginError action creator to remove login error message", () => {
        loginViewInstance.onDismissHandler();
        expect(mockRemoveLoginError).toHaveBeenCalled();
    });

    it("onSubmitHandler dispatches loginFetch action creator when called", () => {
        loginViewInstance.onSubmitHandler({ preventDefault: () => null });
        expect(mockLoginFetch).toHaveBeenCalled();
    });
    it("should map dispatch to props", () => {
        mapDispatchToProps(dispatch).login();
        mapDispatchToProps(dispatch).dismissLoginError();
        expect(dispatch.mock.calls.length).toBe(2);
    });

    it("should map state to props", () => {
        const login = { loginError: "i am an error" };
        expect(mapStateToProps({ login })).toEqual({ loginError: "i am an error" });
    });

});

