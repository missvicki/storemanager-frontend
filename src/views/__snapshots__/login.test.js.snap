// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login view renders loginView component correctly 1`] = `
<Login
  loading={false}
  loginError=""
  onChangeHandler={[Function]}
  onDismissHandler={[Function]}
  onSubmitHandler={[Function]}
  password=""
  role="admin"
  user_name=""
/>
`;
